lib/ruby_llm/chat.rb

```diff
@@ -78,6 +80,112 @@ def with_temperature(temperature)
       self
     end
 
+    # Specifies the response format for the model
+    # @param response_format [Hash, String, Symbol] Either:
+    #   - :json symbol for JSON mode (model outputs valid JSON object)
+    #   - JSON schema as a Hash or JSON string for schema-based output (model follows the schema)
+    # @param strict [Boolean] Whether to enforce the model's support for the requested format
+    # @return [self] Returns self for method chaining
+    # @raise [ArgumentError] If the response_format is not a Hash, valid JSON string, or :json symbol
+    # @raise [UnsupportedJSONModeError] If :json is specified, strict is true, and the model doesn't support JSON mode
+    # @raise [UnsupportedStructuredOutputError] If a schema is specified, strict is true, and the model doesn't support structured output
+    def with_response_format(response_format, strict: true)
```

Comment:
Can you change the method signature to:
```ruby
def with_response_format(strict: true, **response_format)
``

Doing that makes the UI way nicer:
```ruby
chat.with_response_format(type: :number).ask("How old would MLK be today if he were still alive?")
```

Reply:
`with_response_format(response_format, assume_supported: false)`

Reply:
@crmne using destructuring makes the method API much nicer:

```ruby
with_response_format(type = nil, assume_supported: false, **response_schema)
```

Anyways, mixed type variables are a code smell, so this method signature solves that too.